ResourceWatch/Script_geostore_IDs_OW-79.ipynb

```diff
@@ -0,0 +1,1161 @@
+   "cell_type": "code",
+   "execution_count": 83,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import re\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "import geopandas as gpd\n",
+    "import cartoframes as cf\n",
+    "\n",
+    "# Cartoframes docs --> https://carto.com/developers/cartoframes/reference/"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.listdir('./')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['RW_CARTO_KEY', 'RW_CARTO_ACCOUNT']"
+      ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "## env file for gcs upload\n",
+    "env_path = \"../.env\"\n",
+    "with open(env_path) as f:\n",
+    "    env = {}\n",
+    "    for line in f:\n",
+    "        env_key, _val = line.split(\"=\", 1)\n",
+    "        env_value = _val.split(\"\\n\")[0]\n",
+    "        env[env_key] = env_value\n",
+    "        \n",
+    "list(env.keys())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import geojson, json, requests\n",
+    "\n",
+    "def shpToGeojson(s):\n",
+    "    \"\"\"Using a Shapely object, we should build a Geometry object.\"\"\"\n",
+    "    if s.geom_type in ['Polygon', 'Point', 'MultiPoint','MultiPolygon']:\n",
+    "        atts={'geojson': {'type': 'FeatureCollection',\n",
+    "                        'features': [{'type': 'Feature',\n",
+    "                            'properties': {},\n",
+    "                            'geometry': geojson.Feature(geometry=s, properties={}).get('geometry')\n",
+    "                                    }]}}\n",
+    "        return atts\n",
+    "    else:\n",
+    "        raise ValueError('shape object was not of suitable geometry type')\n",
+    "    \n",
+    "def registerGeostore(geojson):\n",
+    "        \"\"\"Register valid geojson to the geostore service. Return the geostore id.\n",
+    "        \"\"\"\n",
+    "        try:\n",
+    "            body = json.loads(json.dumps(geojson))\n",
+    "        except:\n",
+    "            print('Failed to create body')\n",
+    "            return ''\n",
+    "        header= {\n",
+    "                'Content-Type':'application/json'\n",
+    "                }\n",
+    "        url = 'http://api.resourcewatch.org/v1/geostore'\n",
+    "        r = requests.post(url, headers=header, json=body)\n",
+    "        if r.status_code == 200:\n",
+    "            return r.json().get('data', {}).get('id','')\n",
+    "        else:\n",
+    "            print(f'Failed to register geostore. Error {r.status_code}')\n",
```

Comment:
if you can try to avoid ussing print statements for debug, they block the thread. instead, for debugging try to use logger.